src/full/Agda/TypeChecking/Sort.hs

```diff
@@ -85,28 +86,39 @@ hasBiggerSort = void . inferUnivSort
 -- | Infer the sort of a pi type. If we can compute the sort straight away,
 --   return that. Otherwise, return @PiSort a s2@ and add a constraint to
 --   ensure we can compute the sort eventually.
-inferPiSort :: PureTCM m => Dom Type -> Abs Sort -> m Sort
+inferPiSort :: (PureTCM m,MonadConstraint m) => Dom Type -> Abs Sort -> m Sort
 inferPiSort a s2 = do
   s1' <- reduce $ getSort a
-  s2' <- mapAbstraction a reduce s2
+  s2' <- reduce s2
+  s2' <- mapAbstraction a reduce s2'
   -- we do instantiateFull here to perhaps remove some (flexible)
   -- dependencies of s2 on var 0, thus allowing piSort' to reduce
   s2' <- instantiateFull s2'
 
   --Jesper, 2018-04-23: disabled PTS constraints for now,
   --this assumes that piSort can only be blocked by unsolved metas.
+  --Arthur Adjedj, 2023-02-27, Turned PTS back on,
+  --piSort can now be blocked by Leveluniv
+  case piSort' (unEl <$> a) s1' s2' of
+    Right s -> return s
+    Left b -> do
+      let b' = unblockOnAllMetasIn (s1' ,unAbs s2')
```

Comment:
Hm, is it really necessary to use `unblockOnAllMetasIn` here? Perhaps if we replace the calls to `reduce` above by `reduceB` we can use the blocking tags from that to compute a more precise blocking tag. Basically the logic should be that the constraint should be retried whenever `s1'` or `s2'` reduce further, or when the blocker returned by `piSort'` becomes unblocked.